src/Navigation.fs

```diff
@@ -0,0 +1,82 @@
+module Polaris.Navigation
+
+open Fable.Core
+open Fable.Core.JsInterop
+open Fable.React
+open Polaris.Shared
+
+type ItemType = {
+    icon: BundledIcon option
+    label: string
+    url: string
+}
+
+type SectionType = 
+    abstract fill : bool with set

```

Comment:
Why do you use abstracts here, is it because its a 2nd level of props and you can't use lower cased prop names ?
Realisticly, even those other props that are not root level have their own required/non-required props
 (see the example below)
![image](https://user-images.githubusercontent.com/5543043/58222459-19dfdd00-7ce4-11e9-9860-386cbda03049.png)

Perhaps they should follow the same sort of require/non-required pattern with a function that you've been doing ?

Reply:
So, an `ItemType`is a Record and as a Record it has to have all of its properties informed, even if it's an `option`. `SectionType` had no required properties, this tells me that everything is optional, this way it's easier to create an instance of `SectionType`,  just take a look at this as an example
https://github.com/klanthier/fable-polaris/pull/3/files#diff-2ef15224b530477b638d738f0378b515R36

Reply:
You mean Itemtype is not an array of 'react props' for a react component, but rather just an object structure, hence why you would go down that path for these kinds of objects?